quartz/plugins/emitters/contentIndex.ts

```diff
@@ -114,72 +153,202 @@ export const ContentIndex: QuartzEmitterPlugin<Partial<Options>> = (opts) => {
       return graph
     },
     async emit(ctx, content, _resources) {
+      // If we're missing an index file, don't bother with sitemap/RSS gen
+      if (
+        !(
+          opts?.bypassIndexCheck ||
+          content.map(([_, c]) => c.data.slug!).includes("index" as FullSlug)
+        )
+      ) {
+        console.warn(
+          chalk.yellow(`Warning: contentIndex: 
+  content/ folder is missing an index.md. RSS feeds and sitemap will not be generated.
+  If you still wish to generate these files, add:
+    bypassIndexCheck: true,
+  to your configuration for Plugin.ContentIndex({...}) in quartz.config.ts.
+  Don't do this unless you know what you're doing!`),
+        )
+        return []
+      }
+
       const cfg = ctx.cfg.configuration
-      const emitted: FilePath[] = []
-      const linkIndex: ContentIndex = new Map()
+      const emitted: Promise<FilePath>[] = []
+      var indexTree = new Tree<TreeNode>(defaultFeed(), compareTreeNodes)
+
+      // ProcessedContent[] -> Tree<TreeNode>
+      // bfahrenfort: If I could finagle a Visitor pattern to cross
+      //  different datatypes (TransformVisitor<T, K>?), half of this pass could be
+      //  folded into the FeedGenerator postorder accept
+      const detailsOf = ([tree, file]: ProcessedContent): ContentDetails => {
```

Comment:
Seems like this only concerns all folders that contains that is reachable by Quartz

so non-empty folders that doesn't have a `index.md` won't have a RSS file ig.

Reply:
> Seems like this only concerns all folders that contains that is reachable by Quartz

Yes.

> so non-empty folders that doesn't have a `index.md` won't have a RSS file ig.

No, non-empty folders will all have RSS files, unless they have `noRSS: true` in the frontmatter of an `index.md` in the folder. `detailsOf` is called once per content file in the `content` array passed to `emit()`.

Example:

```
content/
├─ afolder/
├─ bfolder/
│  ├─ a.md
│  ├─ b.md
├─ cfolder/
│  ├─ index.md
│  ├─ c.md
├─ index.md
```
produces:
```
public/
├─ bfolder/
│  ├─ a.html
│  ├─ b.html
│  ├─ index.html (default)
├─ bfolder.rss (a, b) (except any that said noRSS)
├─ cfolder/
│  ├─ index.html (user-defined)
│  ├─ c.html
├─ cfolder.rss (c, cfolder/index) (unless cfolder/index.md said noRSS)
├─ index.html
├─ index.xml (index, a, b, c, cfolder/index) (except any that said noRSS)
├─ sitemap.xml/
├─ static/
│  ├─ contentIndex.json
```